### PR TITLE
Add mainloopbteq.c, commandbteq.c, commandbteq.h files

### PR DESCRIPTION
**Pull request by Arman199116**

### CODE REVIEW

File: postgresql-11.5/src/bin/psql/commandbteq.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+typedef enum _dotResult
+{
+	PSQL_CMD_UNKNOWN = 0,		/* not done parsing yet (internal only) */
+	PSQL_CMD_SEND,				/* query complete; send off */
+	PSQL_CMD_SKIP_LINE,			/* keep building query */
+	PSQL_CMD_TERMINATE,			/* quit program */
+	PSQL_CMD_NEWEDIT,			/* query buffer was changed (e.g., via \e) */
+	PSQL_CMD_ERROR				/* the execution of the dot command
+								 * resulted in an error */
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.h:24]: Could you, please, update enum members to be as `BTEQ_CMD_*` as they may conflict with each other.

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 2]
```diff
@@ -0,0 +1,4753 @@
...
+
+#ifdef WIN32
+static void checkWin32Codepage(void);
+#endif
+
+
+
+/*----------
+ * HandleSlashCmds:
+ *
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c:173]: Update function name in comment.

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 3]
```diff
@@ -0,0 +1,4753 @@
+/*
+ * psql - the PostgreSQL interactive terminal
+ *
+ * Copyright (c) 2000-2018, PostgreSQL Global Development Group
+ *
+ * src/bin/psql/command.c
+ */
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c:7]: Update file name and tool name in the comment.

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 4]
```diff
@@ -0,0 +1,4753 @@
...
+	EditableView
+} EditableObjectType;
+
+/* local function declarations */
+static dotResult exec_command(const char *cmd,
+			 PsqlScanState scan_state,
+			 ConditionalStack cstack,
+			 PQExpBuffer query_buf,
+			 PQExpBuffer previous_buf);
+static dotResult exec_command_a(PsqlScanState scan_state, bool active_branch);
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c:61]: Please, set up your coding environment to not have problems with indentation / spaces (line 57-60 and others).

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 5]
```diff
@@ -0,0 +1,4753 @@
...
+#ifdef WIN32
+static void checkWin32Codepage(void);
+#endif
+
+
+
+/*----------
+ * HandleSlashCmds:
+ *
+ * Handles all the different commands that start with '\'.
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c]: Replace '\' with '.'

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 6]
```diff
@@ -0,0 +1,4753 @@
...
+ * Ordinarily called by MainLoop().
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c:175]: This function should be called by `MainLoopBteq` function.

File: postgresql-11.5/src/bin/psql/commandbteq.c [hunk 7]
```diff
@@ -0,0 +1,4753 @@
...
+ *
+ * scan_state is a lexer working state that is set to continue scanning
+ * just after the '\'.  The lexer is advanced past the command and all
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.c:178]: Here as well.

File: postgresql-11.5/src/bin/psql/commandbteq.h [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/*
+ * psql - the PostgreSQL interactive terminal
+ *
+ * Copyright (c) 2000-2018, PostgreSQL Global Development Group
+ *
+ * src/bin/psql/command.h
+ */
+#ifndef COMMAND_H
+#define COMMAND_H
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.h:9]: Update the definition, I wonder this is going not to work properly.

File: postgresql-11.5/src/bin/psql/commandbteq.h [hunk 9]
```diff
@@ -0,0 +1,46 @@
+/*
+ * psql - the PostgreSQL interactive terminal
+ *
+ * Copyright (c) 2000-2018, PostgreSQL Global Development Group
+ *
+ * src/bin/psql/command.h
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/commandbteq.h]: update comment.

File: postgresql-11.5/src/bin/psql/common.c [hunk 10]
```diff
@@ -24,6 +24,7 @@
 
 #include "settings.h"
 #include "command.h"
+#include "commandbteq.h"
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/common.c:27]: do we need this here?

File: postgresql-11.5/src/bin/psql/mainloopbteq.c [hunk 11]
```diff
@@ -0,0 +1,630 @@
+/*
+ * psql - the PostgreSQL interactive terminal
+ *
+ * Copyright (c) 2000-2018, PostgreSQL Global Development Group
+ *
+ * src/bin/psql/mainloop.c
+ */
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/mainloopbteq.c:7]: update comments. this relates to all duplicated files.

File: postgresql-11.5/src/bin/psql/mainloopbteq.c [hunk 12]
```diff
@@ -0,0 +1,630 @@
...
+
+
+/* callback functions for our flex lexer */
+
+/*
+ * Main processing loop for reading lines of input
+ *	and sending them to the backend.
+ *
+ * This loop is re-entrant. May be called by \i command
+ *	which reads input from a file.
```
Comment (maryantonyan) [inline postgresql-11.5/src/bin/psql/mainloopbteq.c]: Remove last two lines from the comment.

**Review by maryantonyan — CHANGES_REQUESTED** [review]

File: postgresql-11.5/src/bin/psql/commandbteq.h (hunk 1 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.h:24]: OK

File: postgresql-11.5/src/bin/psql/commandbteq.c (hunk 2 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.c:173]: Done

File: postgresql-11.5/src/bin/psql/commandbteq.c (hunk 3 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.c:7]: Ok

File: postgresql-11.5/src/bin/psql/commandbteq.c (hunk 4 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.c:61]: Ok

File: postgresql-11.5/src/bin/psql/commandbteq.c (hunk 5 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.c]: Done

File: postgresql-11.5/src/bin/psql/commandbteq.c (hunk 7 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.c:178]: Ok

File: postgresql-11.5/src/bin/psql/commandbteq.h (hunk 8 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.h:9]: Done

File: postgresql-11.5/src/bin/psql/commandbteq.h (hunk 9 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/commandbteq.h]: Done

File: postgresql-11.5/src/bin/psql/common.c (hunk 10 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/common.c:27]: No need, I deleted it

File: postgresql-11.5/src/bin/psql/mainloopbteq.c (hunk 11 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/mainloopbteq.c:7]: Done

File: postgresql-11.5/src/bin/psql/mainloopbteq.c (hunk 12 above)
Comment (Arman199116) [reply postgresql-11.5/src/bin/psql/mainloopbteq.c]: Done

File: postgresql-11.5/src/bin/psql/common.c (hunk 10 above)
Comment (maryantonyan) [reply postgresql-11.5/src/bin/psql/common.c:27]: thanks

**Comment by maryantonyan** [comment]
Arman jan, you need to commit your changes on this branch. so I can see them and approve the PR.
Now I am not able to see any updates mentioned in my comments.

**Review by maryantonyan — APPROVED** [review]
Please, be more careful, you haven't updated some pieces of code regarding to PR notes (such as note related to indentation / spaces). If you have any questions or you don't understand what I mean, feel free to ask what it is about.

**Pull request merged**